src/frontend/src/flows/authenticate/validateDerivationOrigin.ts

```diff
@@ -0,0 +1,76 @@
+import { Principal } from "@dfinity/principal";
+
+export type ValidationResult =
+  | { result: "valid" }
+  | { result: "invalid"; message: string };
+
+export const validateDerivationOrigin = async (
+  authRequestOrigin: string,
+  derivationOrigin?: string
```

Comment:
I'm confused; what is the use case for `derivationOrigin === undefined`? why even call this function?

Reply:
This allows to not having to check whether `derivationOrigin` is `undefined` before calling.

Reply:
Heh, but then it still needs to be checked after calling right? And the value needs to be threaded through. Also in terms of separation of concerns I wonder if this function really cares about this. But 🤷